docs/DeveloperGuide.md

```diff
@@ -215,6 +215,101 @@ New command | Undo command
   * Pros: Will use less memory (e.g. for `delete`, just save the person being deleted).
   * Cons: We must ensure that the implementation of each individual command are correct.
 
+
+### Redo feature (by Jun Cheng)
+
+#### Implementation
+
+The Redo feature was added as a complement to the Undo feature which was done earlier by Zhi Yuan. The addition of this
+feature required the integration of the `RedoCommand` class, which extends from the `Command` class like all other commands.
+The `HistoryStack` class also has new key features to support the `redo` command:
+
+* `HistoryStack#addToRedo(ReadOnlyZooKeepBook)` - Adds a given state of the ZooKeep book into the redo stack.
+* `HistoryStack#removeRecentRedo()` - Removes the most 'recent' update of the ZooKeep book from the redo stack.
+* `HistoryStack#viewRecentRedo()` - Returns (but does not remove) the most 'recent' update of the ZooKeep book.
+* `HistoryStack#clearRedo()` - Clears the future updates of the ZooKeep book stored in the redo stack.
+
+The `RedoCommand` class references some of these methods to accomplish the feature required. Given below is an example
+usage scenario and how the redo mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The ZooKeep book is initialised with the initial state
+given in `data/zookeepbook.json`, and the `HistoryStack` consists of 2 stacks; the history stack and the redo stack,
+each in their respective initial states.
+
+![RedoState0](images/RedoState0.png)
+
+Step 2. The user executes `add n/Harambe...` to add a new animal into the ZooKeep book. The `LogicManager` calls
+`Model#getZooKeepBook()` to retrieve the new state of the book and calls `HistoryStack#addToHistory(ReadOnlyZooKeepBook)`
+as per normal undo protocol.
+
+![RedoState1](images/RedoState1.png)
+
+Step 3. The user then executes `delete 567` which deletes the animal in the book with an ID of 567. Similar to step 2, 
+`LogicManager` will call `Model#getZooKeepBook()` to retrieve the new state of the book and then calls 
+`HistoryStack#addToHistory(ReadOnlyZooKeepBook)` to store this state into the history stack.
+
+![RedoState2](images/RedoState2.png)
+
+Step 4. Now the user thinks that deleting that previous animal was a mistake and wants to restore it by executing `undo`,
```

Comment:
I think no need to re-explain the Undo portion, just refer the reader to the above section. Just skim it over here and focus on Redo.

Reply:
Ok both issues have been fixed and pushed